### PR TITLE
build(material): add bazel rules for all components

### PR DESCRIPTION
**Pull request by jelbourn**
Part of #8369 
cc @DevVersion 

The genrules for transforming the css output are unfortunate but should be able to be removed without a ton of effort.

### CODE REVIEW

**Comment by jelbourn** [comment]
Updated now with CircleCI config to build everything

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       # For some reason, circleci needs the postinstall to be run explicitly.
       # This may be unnecessary once ngcontainer uses nodejs 8
```
Comment (alexeagle) [inline .circleci/config.yml]: it's rules_nodejs that needs to switch, not ngcontainer

File: src/lib/autocomplete/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package(default_visibility=["//visibility:public"])
+load("@angular//:index.bzl", "ng_module")
+load("@io_bazel_rules_sass//sass:sass.bzl", "sass_library")
+load("@io_bazel_rules_sass//sass:sass.bzl", "sass_binary")
```
Comment (alexeagle) [inline src/lib/autocomplete/BUILD.bazel]: you can load these two symbols in one load statement

File: src/lib/autocomplete/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "//src/cdk/bidi",
+    "//src/cdk/collections",
+    "//src/cdk/keycodes",
+    "//src/cdk/portal",
+    "//src/cdk/overlay",
+  ],
+  tsconfig = ":tsconfig-build.json",
+)
+
+
```
Comment (alexeagle) [inline src/lib/autocomplete/BUILD.bazel]: buildozer is available externally, you could format all your BUILD files

File: src/lib/autocomplete/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  deps = ["//src/lib/core:core_scss_lib"],
+)
+
+# TODO(jelbourn): remove this when sass_binary supports specifying an output filename and dir.
+# Copy the output of the sass_binary such that the filename and path match what we expect.
+genrule(
+  name = "autocomplete_css",
+  srcs = [":autocomplete_scss"],
+  outs = ["autocomplete.css"],
+  cmd = "cat $(locations :autocomplete_scss) > $@",
```
Comment (alexeagle) [inline src/lib/autocomplete/BUILD.bazel]: since you have a single src you can
cmd = "cp $< $@"

File: src/lib/autocomplete/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+
+sass_binary(
+  name = "autocomplete_scss",
+  src = "autocomplete.scss",
+  deps = ["//src/lib/core:core_scss_lib"],
+)
+
+# TODO(jelbourn): remove this when sass_binary supports specifying an output filename and dir.
```
Comment (alexeagle) [inline src/lib/autocomplete/BUILD.bazel]: maybe file an issue for that on rules_sass?

File: src/lib/autocomplete/BUILD.bazel (hunk 4 above)
Comment (alexeagle) [reply src/lib/autocomplete/BUILD.bazel]: also, you do this enough that you might want to make a macro around sass_binary...

Comment (jelbourn) [reply src/lib/autocomplete/BUILD.bazel]: The sass_binary rule claims that it has multiple source even though there is just one

File: src/lib/autocomplete/BUILD.bazel (hunk 5 above)
Comment (jelbourn) [reply src/lib/autocomplete/BUILD.bazel]: Done https://github.com/bazelbuild/rules_sass/issues/15

File: src/lib/autocomplete/BUILD.bazel (hunk 3 above)
Comment (jelbourn) [reply src/lib/autocomplete/BUILD.bazel]: Ack, I'll pursue that in a follow-up.

File: .circleci/config.yml (hunk 1 above)
Comment (jelbourn) [reply .circleci/config.yml]: Done

File: src/lib/autocomplete/BUILD.bazel (hunk 2 above)
Comment (jelbourn) [reply src/lib/autocomplete/BUILD.bazel]: Done

**Comment by alexeagle** [comment]
Even if srcs accepts an array I think $< still works if there happens to be
one value?

**Comment by jelbourn** [comment]
@alexeagle 
```
ERROR: /usr/local/google/home/jelbourn/material2/src/lib/button/BUILD.bazel:39:9: in cmd attribute of genrule rule //src/lib/button:button_css: variable '$<' : more than one input file
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@
 anchor_1: &job_defaults
   working_directory: ~/ng
   docker:
-    - image: angular/ngcontainer:0.0.4
+    - image: angular/ngcontainer:0.0.7
```
Comment (alexeagle) [inline .circleci/config.yml]: consider adding buildifier lint

File: .circleci/config.yml [hunk 7]
```diff
@@ -32,9 +32,11 @@ jobs:
 
       - run: bazel run @nodejs//:npm install
       # For some reason, circleci needs the postinstall to be run explicitly.
-      # This may be unnecessary once ngcontainer uses nodejs 8
+      # This may be unnecessary once rules_nodejs uses nodejs 8
```
Comment (alexeagle) [inline .circleci/config.yml]: it does now, upgrade to 0.3.1

File: .circleci/config.yml [hunk 8]
```diff
@@ -32,9 +32,11 @@ jobs:
...
       - run: bazel run @nodejs//:npm run postinstall
       - run: bazel build src/cdk/...
+      - run: bazel build src/lib/datepicker
+      - run: cat /root/.cache/bazel/_bazel_root/049ed6f058a98cba95753eee774cab7a/server/jvm.out
```
Comment (alexeagle) [inline .circleci/config.yml]: debugging?

Comment (jelbourn) [reply .circleci/config.yml]: Yes, I'm in the middle of debugging now

**Comment by jelbourn** [comment]
CircleCI issues are now resolved

**Pull request merged**